server/src/main/java/org/apache/iotdb/db/metadata/MTree.java

```diff
@@ -899,6 +914,70 @@ private void findNodes(MNode node, String path, List<String> res, int targetLeve
     }
   }
 
+  public int getSnapshotLineNumber() {
+    return snapshotLineNumber;
+  }
+
+  public void serializeTo(String snapshotPath, int lineNumber) throws IOException {
+    try (BufferedWriter bw = new BufferedWriter(
```

Comment:
You may need to serialize to a tmp snapshot file first and then rename it to avoid a shutdown when serializing.

Reply:
Updated in latest commit.